lib/puppet/type/acl.rb

```diff
@@ -226,6 +226,9 @@ def initialize(*args)
          to the manifest.
          Reference: #{value.inspect}"
       end
+      if !(value['type'].nil? || value['type'].empty?)
+        warning('Permission `type` is deprecated and has been replaced with perm_type for allow or deny')
+      end
```

Comment:
precedent does not mean we should continue that path for something like this.  This is not an error in the magnitude of the object being completely broken but more along the lines of, 'eh you passed us data we will no longer accept soon'.  So I see them as completely different cases.  If we can catch the deprecation  earlier in validate rather than munge I would prefer in validation.